src/main/java/org/aksw/simba/squirrel/analyzer/mime/RtState.java

```diff
@@ -24,11 +24,11 @@ public RtState(final boolean isFinal, final boolean isError) {
 
     public State transit(final String c) {
         return transitions
-            .stream()
-            .filter(t -> t.isPossible(c))
-            .map(Transition::state)
-            .findAny()
-            .orElseThrow(() -> new IllegalArgumentException("Input not accepted: " + c));
+                .stream()
+                .filter(t -> t.isPossible(c))
+                .map(Transition::state)
+                .findAny()
+                .get();
```

Comment:
Calling `get` can cause a `NoSuchElementException` if there is no element available (which might be possible in a general case). However, your code is not taking care of that case.

Reply:
NoSuchElementException handling has been done